core/shared/src/main/scala/zio/internal/UniqueKey.scala

```diff
@@ -0,0 +1,11 @@
+package zio.internal
+
+/**
+ * This can be used whenever an arbitrary number of unique keys needs to be generated as
+ * this will just use memory location for equality.
+ */
+final class UniqueKey
```

Comment:
I might make these package private to `zio` so users aren't tempted to use them. And so we don't accidentally leak them anywhere.